src/components/widgets/TraceTypeSelector.js

```diff
@@ -0,0 +1,129 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {SearchIcon, ThumnailViewIcon, GraphIcon} from 'plotly-icons';
+import Modal, {ModalContent} from 'components/containers/Modal';
+
+// to be removed when icons are converted to svg
+function slugify(text) {
+  return text
+    .toString()
+    .toLowerCase()
+    .replace(/\s+/g, '-') // Replace spaces with -
+    .replace(/[^\w\-]+/g, '') // Remove all non-word chars
+    .replace(/\-\-+/g, '-') // Replace multiple - with single -
+    .replace(/^-+/, '') // Trim - from start of text
+    .replace(/-+$/, ''); // Trim - from end of text
+}
+
+const Item = ({item, active, columnLength, columnIndex}) => {
+  const isEven = value => value % 2 === 0;
+  const middle = Math.floor(columnLength / 2);
+
+  // for left leaning columns
+  let position = '-right';
+
+  // if we have an even number of columns
+  // we want to have the 2 center rows display their tooltip
+  // in the middle vs left/right
+  if (isEven(columnLength) && columnLength > 3) {
+    if (columnIndex === middle || columnIndex === middle - 1) {
+      position = '';
+    }
+  } else {
+    if (columnIndex === middle) {
+      position = '';
+    }
+  }
+
+  // for right leaning columns
+  if (columnIndex > middle) {
+    position = '-left';
+  }
+  const {label, type} = item;
+  return (
+    <div className={`trace-item${active ? ' trace-item--active' : ''}`}>
+      <div className="trace-item__actions">
+        <a
+          className="trace-item__actions__item"
+          aria-label="Charts like this by Plotly users."
+          data-microtip-position={`top${position}`}
+          role="tooltip"
+          href={`https://plot.ly/feed/?q=plottype:${type}`}
+          target="_blank"
+        >
+          <SearchIcon />
+        </a>
+        <div
+          className="trace-item__actions__item"
+          aria-label="View tutorials on this chart type."
+          data-microtip-position={`top${position}`}
+          role="tooltip"
+        >
```

Comment:
ok, you'd add the tutorial link later? just a reminder :) it should be an `a` tag?
and actually, I think this text _('View tutorials on this chart type') has to be localized.

There's a few places in here where text needs to be localized : )

Reply:
Yeah, I think I am going to abstract these out anyway to they are a part of the config that a user can pass through... enable them, change them, etc